AirCasting/Services/DownloadMeasurmentsService.swift

```diff
@@ -53,13 +53,19 @@ final class DownloadMeasurementsService: MeasurementUpdatingService {
         }
     }
     
-    private func updateForSession(uuid: SessionUUID) {
-        #warning("TODO: change last sync")
-        let syncDate = Date().addingTimeInterval(-100)
+    private func updateForSession(uuid: SessionUUID) {        
+        let session = try? persistenceController.viewContext.existingSession(uuid: uuid)
+        let lastMeasurementTime = session?.allStreams?
+            .compactMap { $0.lastMeasurementTime }
+            .sorted()
+            .last
+        let syncDate = calculateLastSync(sessionEndTime: session?.endTime, lastMeasurementTime: lastMeasurementTime)
+        
         lastFetchCancellableTask = fixedSessionService.getFixedMeasurement(uuid: uuid, lastSync: syncDate, completion: { [removeOldService, persistenceController] result in
             DispatchQueue.main.async {
                 switch result {
                 case .success(let response):
+                    print("From sync date: \(syncDate)")
```

Comment:
Maybe we could use Log.info? 😄

Reply:
ups, sorry, it's left from debugging, I'll remove that ! :D